raiden/transfer/log.py

```diff
@@ -178,12 +201,21 @@ def log(self, state_change):
         serialized_data = self.serializer.serialize(state_change)
         self.storage.write_transaction(serialized_data)
 
+    def log_events(self, events):
+        """ Log the events into the write ahead Log. They are always assumed to be
+        events coming out of the last logged state change"""
+        assert isinstance(events, list)
+
+        id = self.last_state_change_id()
+        for event in events:
+            self.storage.write_state_event(id, self.serializer.serialize(event))
+
     def get_transaction_by_id(self, identifier):
         serialized_data = self.storage.get_transaction_by_id(identifier)
         return self.serializer.deserialize(serialized_data)
 
-    def last_identifier(self):
-        return self.storage.last_identifier()
+    def last_state_change_id(self):
```

Comment:
I don't think the underlying storage id should be exposed, there is no reason for a consumer class of `StateChangeLog` to mess around with it's underlying data store.